### ISSUE TITLE
Open Shell taking 10 seconds to open the start menu after I click it

### ISSUE THREAD

**Issue by warsponge** [Open-Shell/Open-Shell-Menu#1574]
### Describe the bug

Issue started yesterday, i have uninstalled & reinstalled, and deleted all app data, registry keys, restarted my machine, but it still seems to be happening, cant figure out why or what to do to correct it.

if i right click the start menu and go to settings, this settings window opens immediately without issue

To test i installed and ran it in windows sandbox, and it appears to start immediately and functions correctly, so perhaps i am not getting a full uninstall to do a clean install again.

how can i correct this?

### Area of issue

Start menu

### To reproduce

for me to reproduce, I click the start menu, it takes about 10 seconds to open, and then i can carry on using it as normal. it doesnt appear to perform any slower that usual once the start menu has opened

### Open-Shell version

4.4.170

### Windows version

Windows 10 21H2

**Comment by ge0rdi** [comment]
I'd say that there is something added to files/folders that `Open-Shell` enumerates and for some reason it delays the enumeration.

Try to create `Open-Shell` log using this [guide](https://github.com/Open-Shell/Open-Shell-Menu/wiki/How-to-obtain-advanced-Open-Shell-Menu-log). We may be able to see what takes long time.

**Comment by warsponge** [comment]
thanks, i did that, and it did create a log file, which corresponds to the time i tried to open the start menu, however the log file is empty, and if i click edit on the file i get the following error message "The process cannot access the file because it is being used by another process."

unfortunately closing open shell and trying to open the file again still returns the same message

**Comment by ge0rdi** [comment]
Do you happen to install DAEMON Tools Lite recently? Like at the time when your issues started?
Maybe try to uninstall it?

Though it is pretty weird why it should be blocking our logging.

> and ends with this
That means it took 10.4s to open start menu. That is consistent with what you wrote.

> This time it did show some data, although it basically seems to list out my entire user directory, so id rather not paste it here for privacy reasons. it has a total of 903 lines. is this normal?

It is pretty hard to tell if it is normal or not without seeing any data.
`Open-Shell` goes through various folders and enumerates files that should appear in start menu.

You can send me the log to ge0rdi(at)gmx.com if you don't want to attach it here.

**Comment by warsponge** [comment]
No, I've had it installed on my computer for as long as i can remember, i dont really want to install it because i do use it for running .iso files. why its getting involved with this process i have no idea, it is a bit concerning.

> That means it took 10.4s to open start menu. That is consistent with what you wrote.

Ahh i see, so the line number is the time in ms of the process.
in that case then this might be more useful, before i share the whole log file with my info, maybe we can solve without it.

the last 3 lines of the log file are as follows:

>       57: UserAssist: Dropping 'C:\ProgramData\Microsoft\Windows\Start Menu\Programs\Resources\Sublime Text Edit.lnk', pinned to taskbar
   10428: Activate Item, ptr=0000000009C6E7D0, index=13, type=0
   18266: Close Menu, ptr=0000000009C6E7D0

it seems it goes from 57ms to 10428ms between those last lines, i guess there is some problem here, maybe it stumbles on the sublime text editor?

**Comment by ge0rdi** [comment]
This is processing of _recent_/_frequent_ items.

Do you see Sublime in the list of recent apps in start menu?
Try to remove it from there (right click and select Remove from this list) if that helps.

Eventually try to disable showing recent/frequent items in Open-Shell Settings -> Main Menu -> Show recent or frequent programs.

**Comment by warsponge** [comment]
Wow! thankyou, that did the trick.

Sublime wasnt on the recents list so i went in the settings and turned it off to test.
can confirm that must have been it because it opens immediately when i turned it off.
however i do quite like this feature so i played about with it and found that i can have it turned on so long as i have 'Show recent Metro apps' turned off.

thats not a dealbreaker for me so im happy to leave it off, but still, do you have any idea why that would be causing this issue?

**Comment by ge0rdi** [comment]
Hmm, it is hard to tell what could be the issue.

Could you collect [ETW trace](https://github.com/Open-Shell/Open-Shell-Menu/wiki/How-to-obtain-ETW-trace) from time when you press Start button until `Open-Shell` menu shows up?
It should be possible to see what was going on and what we were waiting for.

**Comment by warsponge** [comment]
I have 2 .etl files saved now, can I suggest you update your guide to show where the .etl files are saved, it took me some time to realise they were saved in System32.

What is the best way to send them to you?

**Comment by ge0rdi** [comment]
Trace file is created in the same directory where you start the tracing command.

You can upload it to some cloud storage (google drive, one drive, dropbox, ...) and send me link to it (email is in the guide).

**Comment by Qwerty-Space** [comment]
I'm having a similar issue.

```
      36: Vivaldi.lnk, 0x00410000, 0x00000001
      36: Word.lnk, 0x00410000, 0x00000001
    1351: Activate Item, ptr=000000003BAD4E30, index=19, type=0
    2177: Close Menu, ptr=000000003BAD4E30
```

Removing Word from my start menu doesn't seem to resolve the issue.

**Comment by xuefer** [comment]
in my case it open instantly but after running/idle for some time the popup of start menu is much slower, take about 3 seconds, the longer the slower
